### PR TITLE
Add openapi docs for solver api

### PR DESCRIPTION
**Pull request by vkgnosis**
This starts documenting the solver api outlined by Nic in https://github.com/cowprotocol/services/issues/230 .

In this PR I add only the /qoute route.

I made some design decisions based on what I learned from our Orderbook api and taking into account that this api migth be implemented by external people directly without going through our rust driver implementation.

We only use HTTP status codes 200, 400, 500. It doesn't make sense to (ab)use more status codes for application specific logic because they will not correspond closely enough with their official definition. 200 means ok, 400 means there was something wrong parsing the request, 500 means there was something wrong handling the request. For debugging the server can return some plaintext describing the issue.

For the quote endpoint that means that problems like no liquidity shouldn't be communicated through status codes because they aren't server errors. The request was handled correctly but no quote could be created. I put this into the api in the form of `anyOf` to differentiate "successful quote" from "no quote but also no error".

It didn't make sense to exhaustively list the reasons for no quote being available like we have in the Rust code either, because I'm not sure that in general all potential solvers could agree on these reasons. And users of this api (Autopilot) do not need to know why a quote wasn't available (apart from internal server errors).

### Test Plan

CI now tests validity of this openapi file too

### CODE REVIEW

**Comment by vkgnosis** [comment]
Nic outlined the quote api like this:

```
>> {
>>   sellToken: "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB",
>>   buyToken: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
>>   buyAmount: "1000000000000000000",
>>   // Or, alternatively, for sell quotes:
>>   //sellAmount: "200000000000000000",
>> }

<< {
<<   sellToken: "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB",
<<   buyToken: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
<<   sellAmount: "198107012937109273",
<<   buyAmount: "1000000000000000000",
<<   gas: "200000"
<< }
```

In my version we do not echo the input parameters back in the response. To me this is an anti pattern that adds code and confusion or no benefit.

We use "in_amount" and "kind" in the request and "out_amount" in the response while Nic uses "buy_amount" and "sell_amount". I slightly prefer not changing the layout of the struct based on the order kind because it duplicates the non changing fields but I can also see the case for that approach.

**Comment by nlordell** [comment]
> We use "in_amount" and "kind" in the request and "out_amount" in the response while Nic uses "buy_amount" and "sell_amount". I slightly prefer not changing the layout of the struct based on the order kind because it duplicates the non changing fields but I can also see the case for that approach.

I preferred returning `sell_amount` and `buy_amount` instead of `out_amount` as I found it easier to work with when dealing with external APIs (like 0x, 1Inch and ParaSwap) and find the other causes some more "annoying" code (more match/if statements). This is especially annoying for things like `jq` (like if you want to extract JSON fields from the response) where adding conditional logic is clunky.

Agree that we should have a single input field though.

Either way, don't feel too strongly either way, I just find it much more convenient to work with, and debug, APIs that return both sell and buy amounts.

**Comment by MartinquaXD** [comment]
I would also prefer having a quote response you can work with without requiring the context of the request.
Other than that the PR looks good to me.

File: crates/driver/openapi.yml [hunk 1]
```diff
@@ -39,27 +39,33 @@ components:
...
-        sell_token:
+        sellToken:
           $ref: "#/components/schemas/Address"
-        buy_token:
+        buyToken:
           $ref: "#/components/schemas/Address"
         kind:
           description: |
             - `buy`: in_amount is in buy_token, out_amount is in sell_token
             - `sell`: in_amount is in sell_token, out_amount is in buy_token
```
Comment (MartinquaXD) [inline crates/driver/openapi.yml:49]: This needs to be updated slightly as well now:
```suggestion
            - `buy`: amount is in buy_token
            - `sell`: amount is in sell_token
```

**Review by nlordell — APPROVED** [review]
LGTM

**Comment by nlordell** [comment]
> Btw @nlordell should this API be adapted in any way with relation to your recent PRs about TradeFinding and the like?

I think we could add an additional `/trade` API endpoint that returns calldata so that the price estimation APIs should match the `TradeFinding` interface.

Alternatively, we could also implement it that the trade simulation is done locally in the driver - and have the `autopilot` and `api` trust the price estimates from the driver a bit more (so we would wrap driver price estimation in the `TradeFinderPriceEstimator` - if that makes sense).

I'm still not 100% sure on what the best way forward is (my inclination is to the second option, since trade simulation details may be implementation specific), and we could possibly add in the future a `binding: true` flag to indicate that we want a "hard binding quote that should be settle-able" if we notice an issue (it can potentially even return a Tenderly simulation link, or `eth_call` parameters to that result in a successful run).

I think the PR is fine to merge as is.

**Pull request merged**